### ISSUE TITLE
vector of parameter badly converted to matrix in RKeOps

### ISSUE THREAD

**Issue by AmelieVernay** [getkeops/keops#182]
```r
D <- 2
M <- 3
xl <- c(1, 2, 3)
x <- matrix(xl, 3, 2)
vect <- c(1, 2)

formula = "Sum_Reduction(x+y, 1)"
args = c("x=Vj(2)", "y=Pm(2)")
op1 <- keops_kernel(formula, args)

op1(list(x, vect))
```

Outputs

```
 Error in r_genred(input, param) : 
  [KeOps] Wrong value of the 'vector size' dimension 0 for arg at position 1 : is 1 but should be 2 
```

**Comment by gdurif** [comment]
Origin of the problem: non-matrix input data (in function generated by `keops_kernel`) are converted to matrix-type with `as.matrix`, however vectors (here `vect`) are converted to `n x 1` matrix (or column-vector) instead of `1 x d` matrix (or row-vector).

**Comment by gdurif** [comment]
- [x] Fixed by #183
